client/components/infinite-list/index.jsx

```diff
@@ -341,6 +328,8 @@ export default class InfiniteList extends React.Component {
 	}
 
 	render() {
+		this.scrollHelper.props = this.props;
```

Comment:
We need to manually synchronize `props` between `this` and `this.scrollHelper` and that looks weird and error prone. What if `ScrollHelper` had a more direct reference to the component?

Simple:
```js
class ScrollHelper {
  constructor( component ) {
    this.component = component;
  }
  method() {
    this.component.props.x; // instead of former this.props.x
  }
}
```

More fancy:
```js
class ScrollHelper {
  constructor( component ) {
    this.component = component;
  }
  get props() {
    return this.component.props;
  }
  method() {
    this.props.x; // unchanged
  }
}
```

I believe that in order to make ESLint happy, you'll need to refactor `boundsForRef` somehow? That's also a constructor parameter for `ScrollHelper`, so these two tasks are somewhat related.

Reply:
Another option would be to ditch ScrollHelper entirely and move the logic back into the InfiniteList class. I'm not sure there's much benefit to having them separate.

Reply:
Merged `ScrollHelper` into `InfiniteList`. I don't see it being reused anywhere else, so other than source length, there's no benefit in keeping things split up.